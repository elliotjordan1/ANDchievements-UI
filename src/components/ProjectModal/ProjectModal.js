import React from 'react';
import PropTypes from 'prop-types';
import { 
  Background, 
  ModalWrapper, 
  HeadingText, 
  ProjectBody, 
  BodyText, 
  ListItem, 
  TechListItem, 
  StackedList, 
  LogoWrapper, 
  Image, 
  Icon, 
  ListDescription, 
  TechList 
} from './styles';

const ProjectModal = ({ image, onClick, name, client, blurbOne, blurbTwo, blurbThree, ANDis, techStack, logo }) => {
  return (
    <>
    <Background onClick={onClick}>
      <ModalWrapper onClick={onClick}>
        <Image image={image}>
          <LogoWrapper image={logo}/>
        </Image>
          <ProjectBody>
            <HeadingText><em>{client}</em><br />{name}</HeadingText>
            <BodyText>{blurbOne}</BodyText>
            {blurbTwo ? <BodyText>{blurbTwo}</BodyText> : null}
            {blurbThree ? <BodyText>{blurbThree}</BodyText> : null}
            <BodyText><b>ANDis</b> <br /></BodyText>
            <StackedList>
              {ANDis && ANDis.map((item) => {
                  let andi;
                  try {
                    andi = JSON.parse(item);                    
                  } catch {
                    andi = item;
                  }
                   return (
                    <ListItem key={andi.andiId}>
                      <Icon image={andi.imageURL}/>
                      <ListDescription>
                        {andi.name}
                        <em>{andi.andiProjectRole}</em>
                      </ListDescription>
                    </ListItem>);
                  }
                )}
            </StackedList>
            <TechList>
              {techStack && techStack.map((item) => {
                  let tech;
                  try {
                    tech = JSON.parse(item);
                  } catch {
                    tech = item;
                  }
                  return (
                    <TechListItem key={tech.technologyId}>
                      <Icon image={tech.imageURL}/>
                    </TechListItem>);
              }
              )}
            </TechList>
          </ProjectBody>
      </ModalWrapper>
    </Background>
    </>
  );
};

ProjectModal.propTypes = {
  image: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  client: PropTypes.string.isRequired,
  blurbOne: PropTypes.string.isRequired,
  blurbTwo: PropTypes.string,
  blurbThree: PropTypes.string,
  ANDis: PropTypes.arrayOf(PropTypes.object).isRequired,
  techStack: PropTypes.arrayOf(PropTypes.object).isRequired,
  logo: PropTypes.string.isRequired
};

ProjectModal.defaultProps = {
  blurbTwo: null,
  blurbThree: null
};

export default ProjectModal;
