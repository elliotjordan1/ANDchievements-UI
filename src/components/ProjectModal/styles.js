import styled from 'styled-components';
import theme from '../../global/theme';

export const Background = styled.div`
  width: 100vw;
  height 100vh;
  z-index: 2;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.6);
  top: 48px;
  @media ${theme.media.tablet}{
    top: 64px;
  };
  @media ${theme.media.desktop}{
    top: 88px;
  };
`;

export const ModalWrapper = styled.div`
  width: auto;
  height: 90vh;
  margin: 20px 10px;
  z-index: 1;
  flex-direction: column;
  display: flex;
  @media ${theme.media.tablet}{
    margin: 30px 40px;
    flex-direction: row;
    height: 85vh;
  };
  @media ${theme.media.desktop}{
    margin: 30px 40px;
    height: 85vh;
  };
`;

export const Image = styled.div`
  background-image: ${ ({ image })  => `url(${image})`};
  background-size: cover;
  background-position: center;
  padding-top: 200px;
  width: 100%;
  display: flex;
  @media ${theme.media.tablet}{
    width: 60%;
    padding: 0;
    height: 100%;
    background-color: white;
    background-size: cover;
    background-repeat: no-repeat;
  };
  @media ${theme.media.desktop}{
    background-size: cover;
  }
`;

export const LogoWrapper = styled.div`
  background-image: ${ ({ image })  => `url(${image})`};
  z-value: 3;
  display: flex;
  align-self: flex-end;
  background-size: contain;
  background-repeat: no-repeat;
  width: 175px;
  height: 70px;
  @media ${theme.media.tablet}{
    width: 200px;
    height: 75px;
  };
  @media ${theme.media.desktop}{
    width: 300px;
    height: 100px;
  }
`;

export const HeadingText = styled.div`
  display: block;
  width: 100% - 20px;
  text-transform: uppercase;
  height: min-content;
  font-family: ${theme.fontFamily.primary};
  font-weight: 500;
  font-family: 'Roboto';
  letter-spacing: 0.05em;
  padding: 2px 10px;
  line-height: 1;
  color: ${theme.colours.brand.red};
  background-color: ${theme.colours.brand.grey2};
  font-size: 28px;
  & > em{
    color: ${theme.colours.brand.grey};
    font-size: 20px;
    font-weight: 300;
    font-style: normal;
  }
  @media ${theme.media.tablet}{
    line-height: 1.1;  
    font-size: 28px;
    padding: 10px 10px;
    & > em{
      color: ${theme.colours.brand.grey};
      font-size: 22px;
      font-weight: 300;
      font-style: normal;
    }
  };
  @media ${theme.media.desktop}{
    font-size: 38px;
    line-height: 1.1;
    & > em{
      color: ${theme.colours.brand.grey};
      line-height: 0.4;
      font-size: 32px;
      font-weight: 300;
      font-style: normal;
    }
  };
`;

export const ProjectBody = styled.div`
  line-height: 1.25;
  background-color: white;
  display: flex;  
  width: 100%;
  height: 100%;
  flex-direction: column;
  color: ${theme.colours.brand.grey};
  @media ${theme.media.tablet}{
    width: 40%;
    min-width: 280px;
  };
  @media ${theme.media.desktop}{
    min-width: 530px;
  }
`;

export const BodyText = styled.div`
  max-width: 100%;
  padding: 16px 20px 0 20px;
  font-family: ${theme.fontFamily.primary};
  font-size: 12px;
  font-weight: 300;
  text-align: justify;
  max-height: 9em;
  @media ${theme.media.tablet}{
    font-size: 16px;
  };
  @media ${theme.media.desktop}{
    font-size: 24px;
    line-height: 1.5;
  }
`;

export const StackedList = styled.div`
  list-style-type: none;
  font-family: ${theme.fontFamily.primary};
  font-weight: 400;
  line-height: 2.4;
  letter-spacing: 0.1em;
  color: ${theme.colours.brand.grey};
  margin: 0 20px;
  padding: 0;
  max-width: 50%;
  max-height: 40%;
  flex-direction: column;
  overflow: scroll;
  max-height: 30%;
  flex-wrap: wrap;
  max-width: 100%;
  font-size: 18px;
  @media ${theme.media.tablet}{
    font-size: 20px;
    max-height: 40%;
  };
  @media ${theme.media.desktop}{
    font-size: 28px;
    display: flex;
    overflow: visible;
  }
`;

export const TechList = styled.ul`
  list-style-type: none;
  font-family: ${theme.fontFamily.primary};
  font-weight: 400;
  line-height: 2.4;
  flex-direction: row;
  flex: content;
  display: flex;
  justify-content: space-evenly;
  bottom: 0;
  letter-spacing: 0.1em;
  color: ${theme.colours.brand.grey};
  margin: 0 20px;
  padding: 0;
  bottom: 0;
  max-width: 100%;
  font-size: 18px;
  max-height: 60px;
  @media ${theme.media.tablet}{
    font-size: 22px;
    max-height: none;
  };
  @media ${theme.media.desktop}{
    font-size: 28px;
  }
`;

export const ListItem = styled.div`
  max-width: 100%;
  font-family: ${theme.fontFamily.primary};
  font-size: 12px;
  text-indent: 16px;
  line-height: 1.4;
  display: flex;
  flex-direction: row;
  letter-spacing: 0.05em;
  font-weight: 300;
  height: max-content;
  margin-bottom: 16px;
  margin-left: 8px;
  @media ${theme.media.tablet}{
    font-size: 14px;
  };
  @media ${theme.media.desktop}{
    font-size: 18px;
  }
`;

export const TechListItem = styled.li`
  width: max-content;
  display: flex;
  align-self: flex-end;
  flex-direction: row;
  height: max-content;
  margin-bottom: 16px;
  margin-left: 8px;
`;

export const ListDescription = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  & > em{
    color: ${theme.colours.brand.light};
    font-size: 12px;
    font-weight: 300;
    font-style: normal;
    @media ${theme.media.tablet}{
      font-size: 14px;
    };
    @media ${theme.media.desktop}{
      font-size: 16px;
    }
  }
`;

export const Icon = styled.div`
  background-image: ${ ({ image })  => `url(${image})`};
  background-repeat: no-repeat;
  background-size: contain;
  background-position: center;
  height: 40px;
  width: 40px;
  @media ${theme.media.tablet}{
    height: 50px;
    width: 50px;
  };
  @media ${theme.media.desktop}{
    height: 60px;
    width: 60px;
  }
`;
