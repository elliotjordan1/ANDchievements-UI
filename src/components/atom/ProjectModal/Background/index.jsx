import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
  width: 100vw;
  height 100vh;
  z-index: 2;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.6);
  top: 48px;
  @media ${theme.media.tablet}{
    top: 64px;
  };
  @media ${theme.media.desktop}{
    top: 88px;
  };
`;