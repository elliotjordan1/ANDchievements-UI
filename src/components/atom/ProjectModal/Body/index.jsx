import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
line-height: 1.25;
background-color: white;
display: flex;  
width: 100%;
height: 100%;
flex-direction: column;
color: ${theme.colours.brand.grey};
@media ${theme.media.tablet}{
  width: 40%;
  min-width: 280px;
};
@media ${theme.media.desktop}{
  min-width: 530px;
}
`;