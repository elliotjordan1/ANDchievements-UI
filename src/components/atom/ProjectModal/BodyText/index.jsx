import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
max-width: 100%;
padding: 16px 20px 0 20px;
font-family: ${theme.fontFamily.primary};
font-size: 12px;
font-weight: 300;
text-align: justify;
max-height: 9em;
@media ${theme.media.tablet}{
  font-size: 16px;
};
@media ${theme.media.desktop}{
  font-size: 24px;
  line-height: 1.5;
}
`;