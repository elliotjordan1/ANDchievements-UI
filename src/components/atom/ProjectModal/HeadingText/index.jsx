import styled from 'styled-components';
import theme from '../../../../global/theme';

export default styled.div`
display: block;
width: 100% - 20px;
text-transform: uppercase;
height: min-content;
font-family: ${theme.fontFamily.primary};
font-weight: 500;
font-family: 'Roboto';
letter-spacing: 0.05em;
padding: 2px 10px;
line-height: 1;
color: ${theme.colours.brand.red};
background-color: ${theme.colours.brand.grey2};
font-size: 28px;
& > em{
  color: ${theme.colours.brand.grey};
  font-size: 20px;
  font-weight: 300;
  font-style: normal;
}
@media ${theme.media.tablet}{
  line-height: 1.1;  
  font-size: 28px;
  padding: 10px 10px;
  & > em{
    color: ${theme.colours.brand.grey};
    font-size: 22px;
    font-weight: 300;
    font-style: normal;
  }
};
@media ${theme.media.desktop}{
  font-size: 38px;
  line-height: 1.1;
  & > em{
    color: ${theme.colours.brand.grey};
    line-height: 0.4;
    font-size: 32px;
    font-weight: 300;
    font-style: normal;
  }
};
`;