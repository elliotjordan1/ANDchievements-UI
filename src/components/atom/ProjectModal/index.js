import Background from './Background';
import Body from './Body';
import BodyText from './BodyText';
import HeadingText from './HeadingText';
import Image from './Image';
import LogoWrapper from './LogoWrapper';
import Wrapper from './Wrapper';

export {
  Background,
  Body,
  BodyText,
  HeadingText,
  Image,
  LogoWrapper,
  Wrapper
}