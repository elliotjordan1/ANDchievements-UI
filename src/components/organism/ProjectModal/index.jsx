/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';

import { 
  Background, 
  Wrapper, 
  HeadingText, 
  Body, 
  BodyText, 
  LogoWrapper, 
  Image
} from '../../atom/ProjectModal';

import { TechListContainer, ANDiListContainer } from '../../atom';
import { ANDiListItem, TechListItem } from '../../molecule';

const ProjectModal = ({project, onClick}) => {
  const { imageurl, name, client, blurb_one, blurb_two, blurb_three, andis, techstack, clientlogourl } = project;

  const andiListComponent = andis && andis.map((item) => {
    const andi = JSON.parse(item);

     return (<ANDiListItem andi={andi} key={andi.andiid} />);
    });

  const techStackComponent = techstack && techstack.map((item) => {
    const tech = JSON.parse(item);

    return (
      <TechListItem tech = {tech} key = {tech.technologyid}/>);
  });

  return (
    <Background onClick={onClick}>
      <Wrapper>
        <Image image={imageurl}>
          <LogoWrapper image={clientlogourl}/>
        </Image>
          <Body>
            <HeadingText><em>{client}</em><br />{name}</HeadingText>
            <BodyText>{blurb_one}</BodyText>
            {blurb_two ? <BodyText>{blurb_two}</BodyText> : null}
            {blurb_three ? <BodyText>{blurb_three}</BodyText> : null}
            <BodyText><b>ANDis</b> <br /></BodyText>
            <ANDiListContainer>
              {andiListComponent}
            </ANDiListContainer>
            <TechListContainer>
              {techStackComponent}
            </TechListContainer>
          </Body>
      </Wrapper>
    </Background>
  );
};

ProjectModal.propTypes = {
  project: PropTypes.objectOf(PropTypes.shape).isRequired,
  onClick: PropTypes.func.isRequired
};

export default ProjectModal;
