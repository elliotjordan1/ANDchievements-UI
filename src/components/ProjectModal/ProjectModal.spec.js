import React from 'react';
import { render } from '@testing-library/react';
import ProjectModal from './ProjectModal';
import mockData from '../../assets/mockData.json';

const project = mockData.projects[0];

const mockFunc = jest.fn;

describe('Header component', () => {
  it('renders correctly with mock project', () => {
    const wrapper = render(<ProjectModal 
      image={project.imageURL} 
      name={project.name} 
      client={project.client} 
      blurbOne={project.blurb} 
      ANDis={project.ANDis} 
      techStack={project.techStack} 
      logo={project.clientLogoURL} onClick={mockFunc}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
