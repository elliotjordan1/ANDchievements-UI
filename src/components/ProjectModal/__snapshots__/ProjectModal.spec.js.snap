// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly with mock project 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Background-sc-1txlban-0 bMhZBG"
      >
        <div
          class="styles__ModalWrapper-sc-1txlban-1 fUXPut"
        >
          <div
            class="styles__LogoWrapper-sc-1txlban-8 hKbJpQ"
          />
          <div
            class="styles__TextWrapper-sc-101ii58-1 dFjYKP"
          >
            <div
              class="styles__HeadingText-sc-1txlban-2 jpaJzw"
            >
              <em>
                Big Boss
              </em>
              <br />
              Best Project
            </div>
            <div
              class="styles__ProjectBody-sc-1txlban-3 gaqXcZ"
            >
              <div
                class="styles__BodyText-sc-1txlban-4 dPXiAG"
              >
                Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of 'de Finibus Bonorum et Malorum' (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, 'Lorem ipsum dolor sit amet..', comes from a line in section 1.10.32.
              </div>
              <div
                class="styles__ListWrapper-sc-1txlban-5 fbJcNH"
              >
                <ul
                  class="styles__StackedList-sc-1txlban-6 ixUetG"
                >
                  ANDis 
                  <br />
                  <li
                    class="styles__ListItem-sc-1txlban-7 hPMTpd"
                  >
                    Elliot Joe Jordan Jordan Joe Jordan Jordan Jordan
                  </li>
                  <li
                    class="styles__ListItem-sc-1txlban-7 hPMTpd"
                  >
                    Damiola Olufemi
                  </li>
                </ul>
                <ul
                  class="styles__StackedList-sc-1txlban-6 ixUetG"
                >
                  Tech Stack 
                  <br />
                  <li
                    class="styles__ListItem-sc-1txlban-7 hPMTpd"
                  >
                    React
                  </li>
                  <li
                    class="styles__ListItem-sc-1txlban-7 hPMTpd"
                  >
                    Nodejs
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Background-sc-1txlban-0 bMhZBG"
    >
      <div
        class="styles__ModalWrapper-sc-1txlban-1 fUXPut"
      >
        <div
          class="styles__LogoWrapper-sc-1txlban-8 hKbJpQ"
        />
        <div
          class="styles__TextWrapper-sc-101ii58-1 dFjYKP"
        >
          <div
            class="styles__HeadingText-sc-1txlban-2 jpaJzw"
          >
            <em>
              Big Boss
            </em>
            <br />
            Best Project
          </div>
          <div
            class="styles__ProjectBody-sc-1txlban-3 gaqXcZ"
          >
            <div
              class="styles__BodyText-sc-1txlban-4 dPXiAG"
            >
              Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of 'de Finibus Bonorum et Malorum' (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, 'Lorem ipsum dolor sit amet..', comes from a line in section 1.10.32.
            </div>
            <div
              class="styles__ListWrapper-sc-1txlban-5 fbJcNH"
            >
              <ul
                class="styles__StackedList-sc-1txlban-6 ixUetG"
              >
                ANDis 
                <br />
                <li
                  class="styles__ListItem-sc-1txlban-7 hPMTpd"
                >
                  Elliot Joe Jordan Jordan Joe Jordan Jordan Jordan
                </li>
                <li
                  class="styles__ListItem-sc-1txlban-7 hPMTpd"
                >
                  Damiola Olufemi
                </li>
              </ul>
              <ul
                class="styles__StackedList-sc-1txlban-6 ixUetG"
              >
                Tech Stack 
                <br />
                <li
                  class="styles__ListItem-sc-1txlban-7 hPMTpd"
                >
                  React
                </li>
                <li
                  class="styles__ListItem-sc-1txlban-7 hPMTpd"
                >
                  Nodejs
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
